### DISCUSSION TITLE
Use multiple subscripts in `graphs.csv`

### DISCUSSION THREAD

**Discussion by LSarribouette** [climateinteractive/SDEverywhere#565] [Q&A] [answered]
Hi,

I have successfully transformed my Vensim model into a running JavaScript app, but I had to simplify my model by suppressing multiple subscripts.

- Is it possible to use multiple subscripts in the `graphs.csv` file ? 
- How many subscripts can I use to call variables to plot in the `graphs.csv` file ? We currently use up to 3 subscripts.
- Also, I see a label "vensim graph" in the .CSV file.  Does it mean that plots built in vensim can be used directly ?

This does not work:
```csv
1,,Graphs,Pop,,,,line,,,,,,,,,,,,0,,,,,,,Pop[China,LV],,line,PopChina,blue,,,PopObs[China,LV],,dotted,PopObsChina,red,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```

whereas this works fine:
```csv
1,,Graphs,Pop,,,,line,,,,,,,,,,,,0,,,,,,,Pop1[China],,line,PopChina,blue,,,PopObs1[China],,dotted,PopObsChina,red,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```
with `Pop1[China] = Pop[China,LV]`, defined in the .mdl model

**Comment by chrispcampbell** [comment, marked as answer]
Hi @LSarribouette, 

> Is it possible to use multiple subscripts in the `graphs.csv` file ?

Yes, the issue you're running into is that in the CSV format, when you have a cell that includes commas, you need to "escape" the text in that cell with double quotes.  (See this [SO answer](https://stackoverflow.com/a/4617967) about different escaping options.)

In your example, you'll need to use `"Pop[China,LV]"` and `"PopObs[China,LV]"` like  this:

```csv
1,,Graphs,Pop,,,,line,,,,,,,,,,,,0,,,,,,,"Pop[China,LV]",,line,PopChina,blue,,,"PopObs[China,LV]",,dotted,PopObsChina,red,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```

Note that `@sdeverywhere/plugin-config` uses CSV format for config files because it's based on the build system for En-ROADS.  CSV was chosen because it allows non-technical (and technical) users to configure the app using Excel (or OpenOffice or Pages or any tool that supports CSV), and those tools generally handle the escaping for you.  The files are awkward to edit by hand in a text editor; you can do it as long as you're ready to deal with the CSV formatting gotchas.  (The [Rainbow CSV](https://marketplace.visualstudio.com/items?itemName=mechatroner.rainbow-csv) extension in VS Code can help.)

> How many subscripts can I use to call variables to plot in the `graphs.csv` file ? We currently use up to 3 subscripts.

You can use up to 8 subscripts.

> Also, I see a label "vensim graph" in the .CSV file. Does it mean that plots built in vensim can be used directly ?

Unfortunately, no.  I should probably remove that column name because I see now that it can be confusing, and it's not actually used by `plugin-config`.  The reason it exists is historical (from En-ROADS) where modelers would first build a rough draft of a graph in Vensim, and then would note that in the CSV file for others to translate it into a polished graph for the app.

**Reply by LSarribouette** [reply]
Thank you very much for your answer!
